File: Programação orientada à objetos/TP2-Imobiliaria-em-java-com-UI/TP2 - Codigos/src/Casa.java
public class Casa extends Imovel{

    private int andares;
    private boolean sala_jantar; 
    public Casa (){}

    public Casa (int id, float valor, String proprietario, String rua, String bairro, String cidade, int numero, int quartos, int banheiros, int andares, Boolean sala_jantar) {    
        super (id, valor, proprietario, rua, bairro, cidade, numero, quartos, banheiros);
        this.andares = andares;
        this.sala_jantar = sala_jantar;
    }

    public int getAndares() {
        return andares;
    }

    
    public void setAndares(int andares) {
        this.andares = andares;
    }

    
    public boolean getSala_jantar() {
        return sala_jantar;
    }

   
    public void setSala_jantar(Boolean sala_jantar) {
        this.sala_jantar = sala_jantar;
    }

    public String toString(){

        return String.format("%s\tCasa: Numero de andares = %d\n%s", super.toString(), getAndares(), "----- \n");
    }
}

File: Programação orientada à objetos/TP1-Imobiliaria-em-c++/Codigo/Casa.cpp
#include "casa.h"

Casa :: Casa (int id, float valor, string proprietario, string rua, string bairro, string cidade, int numero, int quartos, int banheiros, int andares, bool sala_jantar):
    Imovel(id, valor, proprietario, rua, bairro, cidade, numero, quartos, banheiros), 
    andares(andares), 
    sala_jantar(sala_jantar) {}

    Casa :: ~Casa() {}

    int Casa :: getAndares() const{
        return andares;
    }

    void Casa :: setAndares(int andares){
        this->andares = andares;
    } 
    bool Casa :: getSalaJantar() const{
        return sala_jantar;
    }

    void Casa :: setSalaJantar(bool sala_jantar){
        this->sala_jantar = sala_jantar;
    }

    void Casa :: saida (ostream& out) const {
        Imovel :: saida(out);
        out <<'\t' << "Casa: ";
        out << "Numeros andares = " << getAndares() << endl
        << "----- \n" ;
    }


    ostream& operator << (ostream& out, const Casa& C){
        C.saida(out);
        return out;
    }
